muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -1003,8 +990,20 @@ private WorkspaceMaterialReply createRestModel(fi.muikku.plugins.workspace.model
     Long workspaceEntityId = entity.getWorkspaceEntity() != null ? entity.getWorkspaceEntity().getId() : null;
     UserEntity userEntity = entity.getUserSchoolDataIdentifier().getUserEntity();
     Long userId = userEntity != null ? userEntity.getId() : null;
+    
+    // TODO Correct user from correct source?
+    User user = userController.findUserByDataSourceAndIdentifier(entity.getWorkspaceEntity().getDataSource(), entity.getUserSchoolDataIdentifier().getIdentifier());
```

Comment:
I'd use entity.getUserSchoolDataIdentifier().getDataSource for the datasource because the user resides in the one and only datasource only.